go/enclave/l2chain/l2_chain.go

```diff
@@ -893,6 +902,53 @@ func (lc *L2Chain) checkAndStoreBatch(batch *core.Batch) error {
 	return nil
 }
 
+// rollbackL2ToLatestValidBatch is called when the currHead's L1 proof block is no longer on the canonical fork
+// it will scan back through batch chain until it finds a batch associated with the canonical L1 chain
+func (lc *L2Chain) rollbackL2ToLatestValidBatch(newL1Head *types.Block) error {
+	currHead, err := lc.storage.FetchHeadBatch()
+	if err != nil {
+		return fmt.Errorf("l2 rollback after fork: unable to fetch head batch - %w", err)
+	}
+	latestValidBatch := currHead
```

Comment:
this variable should have a small comment.
// at the end of the iteration will point to the latest batch we processed that was not reorganised